addons/website/static/tests/builder/block_tab/snippet_groups.test.js

```diff
@@ -551,3 +551,64 @@ test("Cancel snippet drag & drop over sidebar", async () => {
 
     expect(editableContent).toHaveInnerHTML("");
 });
+
+test("Renaming custom snippets don't make an orm call", async () => {
+    const structureSnippetDesc = {
+        name: "Dummy Section",
+        groupName: "custom",
+        content: `
+        <section data-snippet="s_dummy">
+            <div class="container">
+                <div class="row">
+                    <div class="col-lg-7">
+                        <p>TEST</p>
+                    </div>
+                </div>
+            </div>
+        </section>
+    `,
+        keywords: ["dummy"],
+    };
+    const snippets = {
+        snippet_groups: [
+            '<div name="Custom" data-oe-snippet-id="123" data-o-snippet-group="custom"><section data-snippet="s_snippet_group"></section></div>',
+        ],
+        snippet_structure: [getSnippetStructure(structureSnippetDesc)],
+        snippet_custom: [getSnippetStructure(structureSnippetDesc)],
+    };
+    await setupWebsiteBuilder(
+        `<section data-name="Dummy Section" data-snippet="s_dummy">
+            <div class="container">
+                <div class="row">
+                    <div class="col-lg-7">
+                        <p>TEST</p>
+                        <p><a class="btn">BUTTON</a></p>
+                    </div>
+                </div>
+            </div>
+        </section>
+        `,
+        { snippets }
+    );
+
+    await contains(
+        ".o-website-builder_sidebar .o_snippets_container .o_snippet[name='Custom'] button"
+    ).click();
+    await animationFrame();
+    await contains(
+        ".o_add_snippet_dialog .o_add_snippet_iframe:iframe .o_custom_snippet_edit button.fa-pencil"
+    ).click();
+    expect(".o-overlay-item .modal-dialog:contains('Rename the block')").toHaveCount(1);
+    onRpc("ir.ui.view", "rename_snippet", ({ args }) => true);
+    onRpc("ir.ui.view", "render_public_asset", (args) => {
+        throw new Error("shouldn't make an rpc call");
+    });
```

Comment:
Move this stuff (onRpc) at the beginning. I always define all the setup for the test a the beginning to make it more readable :)